cli_tools/common/domain/interfaces.go

```diff
@@ -66,9 +65,20 @@ type TarGcsExtractorInterface interface {
 	ExtractTarToGcs(tarGcsPath string, destinationGcsPath string) error
 }
 
-// StorageObjectDeleterInterface represents an object that is responsible for deleting GCS objects
-type StorageObjectDeleterInterface interface {
-	DeleteObject(bucket string, objectPath string) error
+// StorageObjectInterface represents GCS object interface
+type StorageObjectInterface interface {
+	GetObject(bucket string, objectPath string) StorageObject
+}
+
+// StorageObject represents GCS Object
+type StorageObject interface {
```

Comment:
Having `StorageObjectInterface` and `StorageObject` interfaces is confusing. `StorageObjectInterface` should be named something like `StorageObjectCreatorInterface` or `StorageObjectGetterInterface`

Additionally, it seems `StorageObjectInterface` is there just for testing so you're able to mock retrieving StorageObjects? If so, you can make it non-exported (lower case).

Reply:
I change the name change. Keeping the interface exported as its used by another package (as discussed)